src/heroic.rs

```diff
@@ -0,0 +1,252 @@
+                    // read list of installed games and call find_game_root for result
+                    if let Ok(ins) = serde_json::from_str::<LegendaryInstalled>(
+                        &std::fs::read_to_string(legendary_installed).unwrap_or_default(),
+                    ) {
+                        ins.0.values().for_each(|game| {
+                            log::trace!("detect_legendary_roots found game {} ({})", game.title, game.game_id);
+                            // process game from GamesConfig
+                            if let Some(sp) = self.find_game_root(
+                                root.path.interpret(),
+                                &game.title,
+                                &game.platform.to_lowercase(),
+                                &game.game_id,
+                            ) {
+                                self.memorize_game_root(&game.title, &sp);
+                            }
+                        });
+                    }
+                } else {
+                    log::trace!(
+                        "detect_legendary_roots no such file '{:?}', legendary probably not used yet... skipping",
+                        legendary_installed
+                    );
+                }
+            }
+        }
+    }
+
+    fn detect_gog_roots(&mut self, root: &RootsConfig) {
+        log::trace!(
+            "detect_gog_roots searching for GOG information in {}",
+            root.path.interpret()
+        );
+
+        // use gog_store/library.json to build map .app_name -> .title
+        let mut game_titles = std::collections::HashMap::<String, String>::new();
+        let gog_library = serde_json::from_str::<GogLibrary>(
+            &std::fs::read_to_string(format!("{}/gog_store/library.json", root.path.interpret())).unwrap_or_default(),
+        );
+        gog_library.unwrap().games.iter().for_each(|game| {
+            game_titles.insert(game.app_name.clone(), game.title.clone());
+        });
+        log::trace!(
+            "detect_gog_roots found {} games in {}",
+            game_titles.len(),
+            format!("{}/gog_store/library.json", root.path.interpret())
+        );
+
+        // iterate over all games found in HEROCONFIGDIR/gog_store/installed.json and call find_game_root
+        let content = std::fs::read_to_string(format!("{}/gog_store/installed.json", root.path.interpret()));
+        if let Ok(installed_games) = serde_json::from_str::<HeroicInstalled>(&content.unwrap_or_default()) {
+            installed_games.installed.iter().for_each(|game| {
+                let game_title = game_titles.get(&game.game_id).unwrap();
+                if let Some(sp) = self.find_game_root(root.path.interpret(), game_title, &game.platform, &game.game_id)
+                {
+                    self.memorize_game_root(game_title, &sp);
+                }
+            })
+        }
+    }
+
+    fn memorize_game_root(&mut self, title: &String, path: &StrictPath) {
```

Comment:
I think there are some places here where the term `root` is used when it should be `prefix`.